app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/VpnActiveNetworkManager.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.service
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.ConnectivityManager.NetworkCallback
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import android.os.Handler
+import androidx.core.os.postDelayed
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.vpn.AppTpVpnFeature
+import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
+import com.duckduckgo.mobile.android.vpn.feature.AppTpFeatureConfig
+import com.duckduckgo.mobile.android.vpn.feature.AppTpSetting
+import com.duckduckgo.mobile.android.vpn.network.util.getActiveNetwork
+import com.duckduckgo.mobile.android.vpn.state.VpnStateMonitor.VpnStopReason
+import com.squareup.anvil.annotations.ContributesMultibinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import logcat.LogPriority
+import logcat.asLog
+import logcat.logcat
+import java.util.concurrent.atomic.AtomicReference
+import javax.inject.Inject
+import javax.inject.Provider
+
+/**
+ * This class is bound to the VPN service lifetime.
+ * It watches for changes in the underlying networks (WIFI and CELL) and decides whether
+ */
+@ContributesMultibinding(VpnScope::class)
+class VpnActiveNetworkManager @Inject constructor(
```

Comment:
Extracted the logic from `NetworkTypeCollector` regarding WIWI<>CELL here to avoid firing unnecessary intents (IPC) that needs to be handled by the service.
We already have a service created, so we don't really need heavy IPCs

I know there's some duplication but I am not worry about it. 
Didn't keep the logic in the `NetworkTypeCollector` because it's instanced from two different dagger scopes, `VpnScope` and `AppScope` and that makes it not possible to inject the Vpn service.